File: 303/a/main.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int N;
    cin >> N;
    
    string S, T;
    cin >> S >> T;
    
    string S1 = S;
    string T1 = T;
    
    for (char& c : S1) {
        if (c == 'o') {
            c = '0';
        } else if (c == 'l') {
            c = '1';
        }
    }
    
    for (char& c : T1) {
        if (c == 'o') {
            c = '0';
        } else if (c == 'l') {
            c = '1';
        }
    }
    
    if (S1 == T1) {
        cout << "Yes" << endl;
    } else {
        cout << "No" << endl;
    }
    
    return 0;
}

File: 303/a/main.py
N = int(input())
S = input()
T = input()

S1 = S.replace('o', '0').replace('l', '1')
T1 = T.replace('o', '0').replace('l', '1')

if S1 == T1:
    print('Yes')
else:
    print('No')
